### DISCUSSION TITLE
Kiali performance benchmarks

### DISCUSSION THREAD

**Discussion by ishitharaj** [kiali/kiali#7346] [Q&A]
I am doing a small research on Kiali performance/scalability. Is there any current benchmark results for Kiali?

I am trying to understand how many namespaces/workloads/istio configs that Kiali can handle in a large scale production environment. Currently, I am trying to generate fake configs, deployments and check how Kiali works. I am doing this on a Kubernetes cluster and Kiali runs in a pod. This might not be the best approach to do such a test. Any suggestions on this? Thanks for your valuable opinion.

**Comment by jshaughn** [comment]
Hi @ishitharaj , we also have an effort going on in this area.  @hhovsepy is leading a benchmarking effort, perhaps you would like to contribute, or at least participate in discussion!  In general we see scalability as two things, performance and usability.  Performance is obviously crucial, but also the ability to usefully visualize large meshes.

**Comment by hhovsepy** [comment]
Hi @ishitharaj , we have an epic which covers all Scalability and Performance testing and improvements tasks. https://github.com/kiali/kiali/issues/7058
Currently I am in a process of a setup performance baselines for Kiali on a OCP cluster (3 masters and 2 workers) with "10" and "50" namespaces with each of them 1 Service (2 workloads and 1 app) and 2 config files.
